beeceej/inflight
# Fix/dont vendor anything

Vendoring dependencies means that projects with library constraints will have trouble using this package. removing vendoring.

To support the ^^ I did the following.

* Add in a `type ObjectKeyFunc func() (string, error)` which allows for user driven object id's (also testable now.)
* add a makefile with a `test` target
* use `github.com/google/go-cmp` instad of reflect.DeepEqual
* add two new tests to check the error case in the `type ObjectKeyFunc func() (string, error)`
